hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -100,19 +99,62 @@ public AbfsOutputStream(
     this.appendBlob = appendBlob;
     this.lastError = null;
     this.lastFlushOffset = 0;
-    this.bufferSize = bufferSize;
-    this.buffer = byteBufferPool.getBuffer(false, bufferSize).array();
     this.bufferIndex = 0;
-    this.writeOperations = new ConcurrentLinkedDeque<>();
-
-    this.maxConcurrentRequestCount = 4 * Runtime.getRuntime().availableProcessors();
-    this.threadExecutor
-        = new ThreadPoolExecutor(maxConcurrentRequestCount,
-        maxConcurrentRequestCount,
-        10L,
-        TimeUnit.SECONDS,
-        new LinkedBlockingQueue<>());
-    this.completionService = new ExecutorCompletionService<>(this.threadExecutor);
+
+    init(abfsConfiguration);
+    buffer = new byte[bufferSize];
+  }
+
+  private void init(final AbfsConfiguration conf) {
+    if (isCommonPoolsInitialised()) {
+      return;
+    }
+
+    initWriteBufferPool(conf);
```

Comment:
As discussed, all implementations not tied to the AbfsOutputStream should reside outside this class.